Guard FileView recursion against file nodes

lobre opened the pull request:
Fixes https://github.com/elementary/code/issues/1654.

Fixes #1662

jeremypw reviewed (commented):
This seems more complicated than necessary? I tried just checking that the item is a FolderItem in original line 224 and it seems to fix the problem.  Is there a structure that fails with just this?

jeremypw commented:
Ah, I see - good point.  I wonder whether we could use `GLib.File.get_relative_path ()` rather than `has_prefix ()` and check whether `null` is returned? Hopefully that avoids needing to check for `DIR_SEPARATOR_S`.

lobre commented:
Good catch! `get_relative_path` is cleaner.

I have updated the PR with that approach.

Also, for optimisation, I have added the optional `target_file` parameter, which lets the initial call create the `GLib.File` once, and then pass the same instance down into each recursive call.

Don't hesitate if you see other possible improvements!
